src/utils/utils.js

```diff
@@ -1,88 +1,112 @@
-exports.isNode = function() {
-	return typeof process !== 'undefined' && process.toString() === '[object process]';
-};
+/**
+ * A regular expression that matches whitespace on either side, but
+ * not in the center of a string
+ *
+ * @type {RegExp}
+ */
+var TRIM_REGULAR_EXPRESSION = /^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g;
+
+/**
+ * Used in typeof comparisons
+ *
+ * @type {String}
+ */
+var OBJECT = 'object';
 
+/**
+ * True if environment is node, false if it's a browser
+ * This seems somewhat inelegant, if anyone knows a better solution,
+ * let's change this (must identify browserify's pseudo node implementation though)
+ *
+ * @public
+ * @type {Boolean}
+ */
+exports.isNode = typeof process !== 'undefined' && process.toString() === '[object process]';
+
+/**
+ * Provides as soon as possible async execution in a cross
+ * platform way
+ *
+ * @param   {Function} fn the function to be executed in an asynchronous fashion
+ *
+ * @public
+ * @returns {void}
+ */
 exports.nextTick = function( fn ) {
-	if( exports.isNode() ) {
+	if( exports.isNode ) {
 		process.nextTick( fn );
 	} else {
 		setTimeout( fn, 0 );
 	}
 };
 
+/**
+ * Removes whitespace from the beginning and end of a string
+ *
+ * @param   {String} inputString
+ *
+ * @public
+ * @returns {String} trimmedString
+ */
 exports.trim = function( inputString ) {
 	if( inputString.trim ) {
 		return inputString.trim();
 	} else {
-		return inputString.replace( trimRegExp, '' );
+		return inputString.replace( TRIM_REGULAR_EXPRESSION, '' );
 	}
 };
 
-var trimRegExp = /^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g;
-
-var OBJECT = 'object';
-
-exports.deepEquals = function( objA, objB ) {
-	var isEqual = true, 
-		next;
-
-	if( objA === null || objB === null ) {
-		return objA === objB;
-	}
-
+/**
+ * Compares two objects for deep (recoursive) equality
+ *
+ * This used to be a significantly more complex custom implementation,
+ * but JSON.stringify has gotten so fast that it now outperforms the custom
+ * way by a factor of 1.5 to 3.
+ *
+ * In IE11 / Edge the custom implementation is still slightly faster, but for
+ * consistencies sake and the upsides of leaving edge-case handling to the native
+ * browser / node implementation we'll go for JSON.stringify from here on.
+ *
+ * Please find performance test results here
+ *
+ * http://jsperf.com/deep-equals-code-vs-json
+ *
+ * @param   {Mixed} objA
+ * @param   {Mixed} objB
+ *
+ * @public
+ * @returns {Boolean} isEqual
+ */
+exports.deepEquals= function( objA, objB ) {
 	if( typeof objA !== OBJECT || typeof objB !== OBJECT ) {
 		return objA === objB;
+	} else {
+		return JSON.stringify( objA ) === JSON.stringify( objB );
```

Comment:
I like, was thinking this would be a good approach too.

It doesnt copy over functions though, and I'm not sure about date objects ( on phone now so can't test )